src/main.cpp

```diff
@@ -2080,6 +2081,47 @@ void PartitionCheck(bool (*initialDownloadCheck)(), CCriticalSection& cs, const
     }
 }
 
+Consensus::VersionBitsConditionChecker checkCSV(Consensus::DEPLOYMENT_CSV);
+Consensus::VersionBitsConditionChecker* const softforkcheck[] = {&checkCSV};
+
+int32_t ComputeBlockVersion(const CBlockIndex* pindexPrev, const Consensus::Params& params)
+{
+    LOCK(cs_main);
+    int32_t nVersion = 0;
+
+    for (unsigned int i = 0; i < sizeof(softforkcheck) / sizeof(softforkcheck[0]); i++) {
+        if (softforkcheck[i]->IsSetFor(pindexPrev, params)) {
+            nVersion |= softforkcheck[i]->Mask(params);
+        }
+    }
+
+    if (nVersion == 0) {
```

Comment:
Ok, I get it, it's the old nodes seeing the "empty versionbits" version which would get (unnecessarily?) be told to upgrade and it doesn't matter whether we introduce a SF with BIP9 or not because the start date will be after the miners start using the new code to produce their new versions.

But I think I still disagree, though.
If in march 15th 95% of the miners are ready to deploy a softfork scheduled for april 1st, I believe old nodes should actually receive the upgrade warning already, even if the message is not accurate.